### DISCUSSION TITLE
[Feature] Local-Only Folders / Upload photos not a part of an auto-backed up folder

### DISCUSSION THREAD

**Discussion by yousefmansy1** [immich-app/immich#7075] [Feature Request]
### The feature

From my understanding on mobile the only way to upload a photo to the server is via a folder that is set to be automatically backed up.
My issue with this is there are times you may want to add a photo/video that is not in my typical camera folder. 
(and I certainly don't want to backup my download folder for example)

Secondarily when adding photos to an album it'd be nice to be able to select which photos I want to upload that isn't necessarily part of my existing server gallery.

I feel like this has is probably been brought up elsewhere but I couldn't find any open issues.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by JW-CH** [comment]
I also brought this issue up on the discord once. I'd love to have three states for the folders in the mobile app. Ignore, Show, Back-Up

Ignore: Do not show in the timeline. Do not back up automatically.
Show: Just show in the timeline with the ability to back up manually. Do not back up automatically.
Back-Up: Show in the timeline and automatically back up to the server.

**Comment by yousefmansy1** [comment]
I agree, to be honest we can just copy the implementation of `app that must not be named` "Photos on device" section.

Within that view you can see the local pictures (with appropriate backup icon) and select individual files to upload (or add to album).

Might be an appropriate place to add the backup folder option as well.

**Reply by yousefmansy1** [reply]
To be honest I'm using both nextcloud and immich (yes, storing everything twice) and another killer feature for me is being a le to organize things as a nested directory structure and viewing this as a native file structure on my desktop.

I think that's out of the scope of this discussion though.

**Comment by leowinterde** [comment]
Sharing photos from the file browser via. share as upload to immich is also missing.

**Reply by MexHigh** [reply]
This would be so handy to just "share" the image from the local image library to upload it (e.g. via Android share intent)

**Comment by OJ7** [comment]
> Secondarily when adding photos to an album it'd be nice to be able to select which photos I want to upload that isn't necessarily part of my existing server gallery.

I ran into this limitation recently. I have users who don't necessarily want to use auto backup, but only upload a subset of media files (e.g. from family events). The workaround is uploading via the mobile web ui each time and then manually adding those to albums.

**Comment by echondropoulos** [comment]
Really disappointed this feature isn't already available on mobile.

I have a large diverse family and I wanted a single app where we could all upload our favorite photos into predefined albums with ease.

Being able to upload a single photo from my phone to a shared album from within the app will achieve this, but Immich does not support this from within the app.

I would really appreciate this to be added!

**Comment by b-Tomas** [comment]
I would very much appreciate this feature. I certainly don't want to backup every single image I get on whatsapp, but every now and then there is one that I'd like to upload.
This feature is available on the _**App-Which-Must-Not-Be-Named**_ and makes it quite easy to access non-backed up folders in the device, containing files that are not shown in the timeline, select multiple files and upload them manually. I believe it is a an ok UX from which Immich may take some inspiration.

Edit: And there is something else missing here I noticed. I think the ability to share files to the Immich app for them to be uploaded would be convenient.

My current workaround is: I create a folder called `to-be-backed-up` in my phone, and when someone sends me some pictures I'd like to save, I copy them into that folder, which is backed up by Immich.

**Comment by SHU-red** [comment]
+1

- Open image (in whatever app)
- Share --> Immich
Backup done

EDIT:
What works as a quick-fix is to open up the images in e.g. Google Files and copy it to the Camera folder
Immich detects and backups the new photos

Enough for me right now but implementing the Android-Share-Function to Immich would be great

**Comment by bo0tzz** [comment]
#11905

**Reply by OJ7** [reply]
Should the above (new) issue be closed as a duplicate instead of this?

**Reply by bo0tzz** [reply]
Whoops, I switched them up. Thank you!

**Reply by mewtwotwo** [reply]
> Whoops, I switched them up. Thank you!

Are there plans to implement this feature?

**Comment by AhmedNSidd** [comment]
A highly needed feature in Immich I'd say, especially for my use case. If family members and I are out on a trip, and we want to share our pictures with each other in a shared album, they shouldn't necessarily have to share their entire local device albums in order to add a select few pictures into the shared album.

Here's how the flow could look (as also alluded to by other commenters in this discussion):
1. Create a shared album with other Immich users who already have accounts.
2. Other Immich users can go directly to the shared album and press the button for adding their photos. There they have the choice to choose pictures from their Immich Library. Immich Library has everything they've uploaded to Immich already. They also have the option to pick the pictures from their local device photos. If they choose to pick photos from their local device, then they'll have the option to select a photo, or multiple photos, and upload them to the shared album.
3. Alternatively, instead of adding the photos to the shared album from the album directly, they can navigate to the local device photos in Immich's library. These photos needn't necessarily be backed up in order for a user to view them. There, they have the option to select a photo, or a group of photos to their Shared Album.

I think alongside with this feature request, we also need to prioritize this feature request as well in order to accomplish the vision highlighted above: https://github.com/immich-app/immich/discussions/2266

**Reply by T0ytoy** [reply]
I would also love the same feature you are describing, for family and friends events, to let those people send photos without having to use the web interface or uploading their whole library!

**Reply by MexHigh** [reply]
I've worked around this by creating a dedicated user for this use case. Under this user I would create (empty) shared albums for each event with uploading enabled. This way, the family member don't even need an account to share their photos.

**Comment by csuno** [comment]
we came from Synology. And with their actual changes on ther apps (also "Synology Photos")... theree will migrate some user from synology to immich... but this is a big missig feature!

Our Family just wants to select and upload special selected pictures from there mobile gallery/library... and dont want to setup the whole Backupmechanism, also even with this it is much to complicate, to share a single photo to a shared album of all!

**Comment by aciulpan** [comment]
Will this feature be eventually added? It's very hard not being able to upload single files

**Comment by scoryllo** [comment]
I also hope it comes soon, it's the only feature missing for us.

**Comment by LaVINCENTgit** [comment]
Please, implement the possibility to select photos from source device and upload them in the immich server but directly in a specific album... Thanks!

**Comment by errsby** [comment]
Insanity that this hasn't been added for a whole year, a simple upload button with that allows you to select images to be uploaded would go a long way :(

**Comment by yousefmansy1** [comment]
So it looks like as of `1.136.0` the beta timeline feature actually supports this feature.

Minor feedback, any chance we can add an option to the top to set the sorting order (recently edited, name, etc) and to switch between list and grid view?

**Comment by csuno** [comment]
i cant see an Option to manually add Pictures!
ios AppVersion 2.4.1 build 239
Server 2.4.1

can you tell us, how it is closed as "resolved"?

**Reply by alextran1502** [reply]
Go to Library > On This Device > Select any local album, from that view, you can select any photo and press Upload

**Reply by csuno** [reply]
thanks! Didnt know this!

**Reply by LaVINCENTgit** [reply]
Hi Alex, should be fantastic the possibility to seletct in which album do the upload :) thanks
